File: ui/src/api/history.ts
import axios from "axios";

const history_api = axios.create({
    baseURL: "http://localhost:8000/history",
    timeout: 10000,
});

export async function RecordHistory(search_query: string, time: string, num: number) {
    try {
        const response = await history_api.get("/record_history", {
            params: {
                search_query: search_query,
                time: time,
                num: num
            }
        });
        return response.data;  // 返回实际数据而不是整个响应对象
    } catch (error) {
        console.error("RecordHistory error:", error);
        throw error;  // 重新抛出错误以便调用者处理
    }
}

export async function GetAllHistory() {
    try {
        const response = await history_api.get("/get_history")
        return response.data;
    }
    catch (error) {
        console.error("GetAllHistory error:", error);
        throw error;  // 重新抛出错误以便调用者处理
    }
}

export async function DeleteHistory(id: number) {
    try {
        const response = await history_api.get("/remove_history", {
            params: {
                id: id
            }
        });
        return response.data;
    } catch (error) {
        console.error("DeleteHistory error:", error);
        throw error;  // 重新抛出错误以便调用者处理
    }
}

export async function DeleteAllHistory() {
    try {
        const response = await history_api.get("/remove_all_history");
        return response.data;
    } catch (error) {
        console.error("DeleteAllHistory error:", error);
        throw error;  // 重新抛出错误以便调用者处理
    }
}

File: backend/routers/history.py
# =============== description ===============
# 这里是历史记录的接口
# ===========================================
from fastapi import APIRouter
from backend.services.history_service import save_history,get_all_history,delete_history,delete_all_history

router = APIRouter(
    prefix="/history",
    tags=['history']
)

@router.get("/record_history")
async def record_history(search_query:str,time:str,num:int):
    result = save_history(search_query=search_query,time=time,num=num)
    return result

@router.get("/get_history")
async def get_history():
    results = get_all_history()
    return results

@router.get("/remove_history")
async def remove_history(id:int):
    result = delete_history(id=id)
    return result

@router.get("/remove_all_history")
async def remove_all_history():
    result = delete_all_history()
    return result